File: src/main/java/com/selqet/momrecipebook/controller/IngredientController.java
package com.selqet.momrecipebook.controller;

import com.selqet.momrecipebook.entity.IngredientEntity;
import com.selqet.momrecipebook.model.IngredientDto;
import com.selqet.momrecipebook.repository.IngredientRepository;
import com.selqet.momrecipebook.service.impl.IngredientServiceImpl;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

//TODO Check every method for exceptions

@RestController
@RequestMapping("/ingredients")
public class IngredientController {
    private IngredientServiceImpl ingredientService;

    public IngredientController(IngredientServiceImpl ingredientService) {
        this.ingredientService = ingredientService;
    }

    @GetMapping("/all")
    public ResponseEntity<List<IngredientDto>> getAllIngredients() {
        List<IngredientDto> allIngredients = ingredientService.getAllIngredients();
        return new ResponseEntity<>(allIngredients, HttpStatus.OK);
    }

    @PostMapping("/add")
    public ResponseEntity<Object> addIngredient(@RequestBody IngredientDto ingredient) {
        ingredientService.addIngredient(ingredient);
        return new ResponseEntity<>(HttpStatus.CREATED);
    }

    @DeleteMapping("/delete/{id}")
    public ResponseEntity<Object> deleteIngredient(@PathVariable Long id) {
        ingredientService.deleteIngredient(id);
        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
    }

    //TODO !!!Already exists exception
    @PutMapping("/update/{id}")
    public ResponseEntity<Object> updateIngredient(@PathVariable Long id, @RequestBody IngredientDto ingredient) {
        ingredientService.updateIngredient(id, ingredient);
        return new ResponseEntity<>(HttpStatus.CREATED);
    }
}

File: src/main/kotlin/com/example/momrecipebook/controller/IngredientController.kt
package com.example.momrecipebook.controller

import com.example.momrecipebook.dto.IngredientDto
import com.example.momrecipebook.service.IngredientService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class IngredientController(
    private val service: IngredientService,
) {
    @GetMapping
    fun getAll(): List<IngredientDto> = service.getAll()

    @PostMapping
    fun create(@RequestBody dto: IngredientDto): Long = service.create(dto)

    @DeleteMapping("/{name}")
    fun delete(@PathVariable name: String) {
        service.deleteByName(name)
    }
}